enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md

```diff
@@ -0,0 +1,253 @@
+---
+title: cluster-logging-es-rollover-data-design
+authors:
+  - "@jcantril"
+reviewers:
+  - "@bparees"
+  - "@ewolinetz"
+  - "@richm"
+  - "@portante"
+  - "@pavolloffay"
+  - "@sichvoge"
+approvers:
+  - "@ewolinetz"
+  - "@richm"
+creation-date: 2019-11-04
+last-updated: 2019-11-04
+status: implementable
+---
+
+# Cluster Logging Elasticsearch Rollover Data Design
```

Comment:
It isn't really about just rollover data design - it is about using rollover + single indices in conjunction with DLS for access control/multi-tenancy

Reply:
This proposal is strictly about modifying the data design and introducing rollover.  It is correct that in order for us to full-fill our multitenant requirements we will need to implement DLS but that is to be addressed in a separate proposal. I would expect we do both or neither.

Reply:
Seems like you need to explicitly link this design to be depend on the DLS design being accepted, otherwise this design is a non-starter.

Reply:
@portante that is not correct.  Rollover addresses our scalability and performance concerns but not multi-tenancy requirements.  It's possible that we could design something that sits in front of ES to restrict access which is orthogonal to roll over.

Reply:
> I would expect we do both or neither.

Even without DLS the roll-over would be still great improvement. If DLS is not in place we should still push for use of roll-over.

Reply:
> @portante that is not correct. Rollover addresses our scalability and performance concerns but not multi-tenancy requirements. It's possible that we could design something that sits in front of ES to restrict access which is orthogonal to roll over.

If you have N namespaces, you still have N indices if the size of the data in those indices does not meet the rollover limits.  The scale of the number of indices is still based on namespaces, not size/time.

So this might alleviate the issue in some cases, but without DLS, or some other implementation that addresses multi-tenancy, it is not clear how this can be a "design" that would be accepted and implemented in the product.